test/unit/postprocessing/test_curve_utilities.py

```
@@ -55,75 +55,82 @@ def test_interpolate_curve():
-                }
-            ),
+            {
+                "x": [0.0, 0.2, 0.3, 0.5, 1.0],
+                "y": [0.0, 0.3, 0.35, 0.7, 1.0],
+                "operation": ["i", "r", "r", "e", "l"],
+            },
             [0, 1, 3, 4],
         ),
         (
-            pd.DataFrame(
-                {
-                    "x": [0, 0.1, 0.2, 0.5, 1],
-                    "y": [0, 0.3, 0.5, 0.9, 1],
-                    "operation": ["i", "r", "r", "e", "l"],
-                }
-            ),
+            {
+                "x": [0.0, 0.1, 0.2, 0.5, 1.0],
+                "y": [0.0, 0.3, 0.5, 0.9, 1.0],
+                "operation": ["i", "r", "r", "e", "l"],
+            },
             [0, 1, 2, 3, 4],
         ),
         (
-            pd.DataFrame(
-                {
-                    "x": [0, 0.2, 0.3, 0.5, 1],
-                    "y": [0, 0.3, 0.8, 0.82, 1],
-                    "operation": ["i", "r", "r", "e", "l"],
-                }
-            ),
+            {
+                "x": [0.0, 0.2, 0.3, 0.5, 1.0],
+                "y": [0.0, 0.3, 0.8, 0.82, 1.0],
+                "operation": ["i", "r", "r", "e", "l"],
+            },
             [0, 2, 4],
         ),
     ],
 )
-def test_convex_hull(base_points, expected_remaining_indices):
+def test_convex_hull(constructor, data, expected_remaining_indices):
+    base_points = constructor(data)
     convex_hull = _filter_points_to_get_convex_hull(base_points)
-    pd.testing.assert_frame_equal(
-        base_points.iloc[expected_remaining_indices].reset_index(drop=True), convex_hull
-    )
+
+    expected = nw.maybe_reset_index(
+        nw.from_native(base_points, eager_only=True, pass_through=False)[
+            expected_remaining_indices
+        ]
+    ).to_native()
+    assert expected.equals(convex_hull)
 
 
-def test_calculate_tradeoff_points():
-    data = pd.DataFrame(
+def test_calculate_tradeoff_points(request, constructor):
+    if "pyarrow_table" in str(request):
```

@FBruzzesi Should we raise an exception early on, in the two outermost function of this module (_tradeoff_curve, _interpolate_curve), if we detect that the dataframe that was passed is a pyarrow table since we do not support that yet ?

---

I checked the pyarrow documentation and storing a custom class in a pyarrow table column might not be so straightforward although possible with extension types.

Now that you mention this, ideally I would move the `TypeError` at the `ThresholdOptimizer.fit` method level, which is the only method that uses both `_tradeoff_curve` and `_interpolate_curve`. In the tests we can keep the xfail at the lower level for now, until we add Narwhals support for `ThresholdOptimizer`. What do you think?